src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -121,6 +98,56 @@ case class CsvRelation protected[spark] (
     new InterpretedMutableProjection(casts, startSchema)
   }
 
+  private def readCSV(
+     file: RDD[String],
+     header: Seq[String],
+     schemaFields: Seq[StructField],
+     projection: MutableProjection,
+     row: GenericMutableRow) = {
+
+    val dataLines = if (useHeader) {
+      file.mapPartitionsWithIndex({
+        case (partitionIndex, iter) => if (partitionIndex == 0) iter.drop(1) else iter
```

Comment:
do you have any examples where this happens? also, what are the chances that if a dataset does have repeated header lines, they are exactly identical? what if there is an extra space or a lowercase/uppercase mismatch or something else that will not be caught by the filter? how does one include a "data" line in the file that happens to look like the header?